code/modules/projectiles/guns/energy/modular.dm

```diff
@@ -0,0 +1,199 @@
+/obj/item/weapon/gun/energy/laser/prototype
+	name = "laser prototype"
+	desc = "A custom prototype laser weapon."
+	icon_state = "laser"
+	item_state = "laser"
+	fire_sound = 'sound/weapons/Laser.ogg'
+	slot_flags = SLOT_BELT|SLOT_BACK
+	w_class = 3
+	force = 10
+	origin_tech = list(TECH_COMBAT = 3, TECH_MAGNET = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 2000)
+	can_turret = 0
+	zoomdevicename = null
+	var/origin_chassis
+	var/gun_type
+	var/list/gun_mods
+	var/obj/item/laser_components/capacitor/capacitor
+	var/obj/item/laser_components/focusing_lens/focusing_lens
+	var/chargetime
+	var/is_charging
+
+/obj/item/weapon/gun/energy/laser/prototype/examine()
```

Comment:
Might be good to add a distance check. people likely wouldn't be able to identify what parts they were unless they were within a tile or two